player-computer.js

```diff
@@ -0,0 +1,171 @@
+const readline = require('readline');
+const rl = readline.createInterface(process.stdin, process.stdout);
+let minNum = 1
+let maxNum = 100
+function ask(questionText) {
+    return new Promise((resolve, reject) => {
+        rl.question(questionText, resolve);
+    });
+}
+
+// Function for a random number - computer number
+function randomComNum(min, max) {
+    return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+//Global variable to keep track of number of tries
+let count = 1;
+
+// Binary search function
+function randomNum(min, max) {
+    return Math.floor((min + max) / 2)
+}
+
+async function initialize() {
+    console.log('We are going to play a game in which we have to think of a number and the other will have to guess it!')
+
+    let response = await ask('Would you want me(computer) to guess your secret number?\nOr would you(human) would like to guess my secret number?. Computer guesses(C) or Human guesses(H)')
+
+    if (response.toLowerCase() === "c") {
+        start()
+    } else if (response.toLowerCase() === "h") {
+        startComputer()
+    }
+
+    // making sure we are getting a valid response
+    while (response !== "c" && response !== "h") {
+        console.log('Please enter a valid response')
+        response = await ask("Computer guesses(C) or Human guesses(H)")
+    }
+}
+
+initialize();
+
+// computer guesses
+async function start() {
+    console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
+
+    let maxRange = await ask("First, let's determine your range! from 1 to ...? (Enter a number greater than 1)")
+
+    // making sure the user enters an actual number.
+    while (isNaN(maxRange)) {
+        maxRange = await ask("That's not a valid response! Give me an actual number! (Enter a number greater than 1)")
+    }
+
+    let secretNumber = await ask(`And now... think of a number within 1 and ${maxRange}! What is your secret number?\nI won't peek, I promise...\n`);
+    console.log('You entered: ' + secretNumber);
+
+    // making sure the user enters an actual number.
+    while (isNaN(secretNumber)) {
+        secretNumber = await ask("That's not a valid response!Pick an actual number!")
+    }
```

Comment:
The only bug that I found was that in here, you can submit a negative number for them which would make the game unable to be played completed but the computer still wouldn't call you a cheater. 

changing your while loop form `while (isNaN(maxRange))` to something like `while (isNaN(maxRange) && maxRange <= 0)` could be a way to prevent the user from guessing a negative number